### PR TITLE
Ensure old host/batch system configs don't conflict with dummy-local mode

### PR DESCRIPTION
**Pull request by wxtim**
This is a small change not directly addressing any issue, although it touches on #3482.

@oliver-sanders  observed the following issue:
> Just spotted a platforms issue. When you run with --mode=dummy it adds platform=dummy. If the suite contains stuff like [remote]host or [job]batch system this will result in mixed syntax which causes the upgrader to fail and raise a PlatformLookupError.

This PR attempts to remedy this bug.

### CODE REVIEW

File: tests/functional/validate/76-host-to-platform-upgrade-dummy-mode.t [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
```
Comment (oliver-sanders) [inline tests/functional/validate/76-host-to-platform-upgrade-dummy-mode.t:1]: This shouldn't be in the validate tests, move in with the run tests or put somewhere else.

**Review by oliver-sanders — APPROVED** [review]

File: tests/functional/validate/76-host-to-platform-upgrade-dummy-mode.t (hunk 1 above)
Comment (hjoliver) [reply tests/functional/validate/76-host-to-platform-upgrade-dummy-mode.t:1]: (maybe a new "platforms" test sub-dir?)

Comment (wxtim) [reply tests/functional/validate/76-host-to-platform-upgrade-dummy-mode.t:1]: how about a `upgrade7to8` folder - if used correctly it'll help us know whether we've deprecated the rather involved upgrade syntax?

**Review by hjoliver — APPROVED** [review]
LGTM, tests as working :+1:

**Pull request merged**